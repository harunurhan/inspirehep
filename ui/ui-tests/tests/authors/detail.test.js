const { routes, selectors } = require('../../utils/constants');
const { createPollyInstance } = require('../../utils/polly');
const { login, logout } = require('../../utils/user');
const {
  takeScreenShotForDesktop,
  takeScreenShotForMobile,
} = require('../../utils/screenshot');

describe('Author Detail', () => {
  beforeAll(async () => {
    await login();
  });

  it('should match image snapshot for an Author', async () => {
    await page.setRequestInterception(true);
    const polly = createPollyInstance('AuthorDetail');

    await page.goto(routes.public.authorDetail983328, {
      waitUntil: 'networkidle0',
    });
    await page.waitFor(selectors.searchResults);

    const desktopSS = await takeScreenShotForDesktop(page);
    // FIXME: citations by year graph is rendered incorrectly sometimes, therefore `failureThreshold`
    expect(desktopSS).toMatchImageSnapshot({
      failureThreshold: '0.071',
      failureThresholdType: 'percent',
    });

    const mobileSS = await takeScreenShotForMobile(page);
    expect(mobileSS).toMatchImageSnapshot();

    await polly.stop();
  });

  afterAll(async () => {
    await logout();
  });
});
