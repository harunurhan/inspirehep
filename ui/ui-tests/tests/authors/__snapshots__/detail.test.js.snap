// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Detail sets "author brief" as meta description 1`] = `"Humboldt U., Berlin and Leipzig U.. hep-lat"`;
